File: page4.js
const gameBoard = document.getElementById('game-board');
const nextBtn = document.getElementById('next-btn');

let hasFlippedCard = false;
let lockBoard = false;
let firstCard, secondCard;
let matchedPairs = 0;

const images = [
    'game1.jpg', 'game2.jpg', 'game3.jpg', 
    'game4.jpg', 'game5.jpg', 'game6.jpg'
];

const cardData = [...images, ...images];
cardData.sort(() => Math.random() - 0.5);

function createBoard() {
    cardData.forEach(img => {
        const card = document.createElement('div');
        card.classList.add('memory-card');
        card.dataset.image = img;
        card.innerHTML = `
            <div class="front-face"><img src="${img}"></div>
            <div class="back-face">❤️</div>
        `;
        card.addEventListener('click', flipCard);
        gameBoard.appendChild(card);
    });
}

function flipCard() {
    if (lockBoard || this === firstCard) return;
    this.classList.add('flip');
    if (!hasFlippedCard) {
        hasFlippedCard = true;
        firstCard = this;
        return;
    }
    secondCard = this;
    checkForMatch();
}

function checkForMatch() {
    let isMatch = firstCard.dataset.image === secondCard.dataset.image;
    if (isMatch) {
        matchedPairs++;
        if (matchedPairs === 6) nextBtn.style.display = 'inline-block';
        resetBoard();
    } else {
        lockBoard = true;
        setTimeout(() => {
            firstCard.classList.remove('flip');
            secondCard.classList.remove('flip');
            resetBoard();
        }, 1000);
    }
}

function resetBoard() {
    [hasFlippedCard, lockBoard] = [false, false];
    [firstCard, secondCard] = [null, null];
}

createBoard();

File: page4.dart
const photos = [
    'photo1.jpg', 
    'photo2.jpg', 
    'photo3.jpg',
    'photo1.jpg', 
    'photo2.jpg', 
    'photo3.jpg'
];

let board = document.getElementById('gameBoard');
let flippedCards = [];
let matchedCount = 0;

// Shuffle photos
const shuffledPhotos = photos.sort(() => 0.5 - Math.random());

// Create Cards
shuffledPhotos.forEach((imgSrc, index) => {
    const card = document.createElement('div');
    card.classList.add('card');
    card.dataset.name = imgSrc;
    card.innerHTML = `
        <div class="card-back">❤️</div>
        <div class="card-front"><img src="${imgSrc}"></div>
    `;
    card.addEventListener('click', flipCard);
    board.appendChild(card);
});

function flipCard() {
    if (flippedCards.length < 2 && !this.classList.contains('flipped')) {
        this.classList.add('flipped');
        flippedCards.push(this);

        if (flippedCards.length === 2) {
            checkMatch();
        }
    }
}

function checkMatch() {
    const [card1, card2] = flippedCards;
    if (card1.dataset.name === card2.dataset.name) {
        matchedCount += 2;
        flippedCards = [];
        if (matchedCount === photos.length) {
            setTimeout(() => {
                document.getElementById('winMessage').classList.remove('hidden');
            }, 500);
        }
    } else {
        setTimeout(() => {
            card1.classList.remove('flipped');
            card2.classList.remove('flipped');
            flippedCards = [];
        }, 1000);
    }
}